### ISSUE TITLE
When handling a scatter query vtgate should ensure consistent schema

### ISSUE THREAD

**Issue by setassociative** [vitessio/vitess#4669]
#### Overview of the Issue

If a table is in the middle of a schema migration it's possible for a scattered `SELECT *` query to return an inconsistent schema depending on the state of each tablet. When aggregating results we should ensure that each result contains the same set of fields and return an error if not.

**Caveat**: I'm not positive this is not already handled but based on commentary in [this comment](https://github.com/vitessio/vitess/pull/4660#discussion_r259947389) I don't believe it is.

#### Reproduction Steps

1. Begin making a schema change
2. Issue a scatter `select *` before the schema is applied to all shards

**Comment by indera-shsp** [comment]
@setassociative Did you guys deploy this patch to production?

https://github.com/tinyspeck/vitess/commit/3d3f6609138ca38c34531f3ecef5ab2fb4f52af9
I am encountering this issue when trying to connect to the Vstream

```
panic: runtime error: index out of range [4] with length 4

goroutine 30 [running]:
vitess.io/vitess/go/sqltypes.MakeRowTrusted(0xc0001be220, 0x4, 0x4, 0xc000169e50, 0x8, 0xdd0840, 0x1)
	/go/pkg/mod/vitess.io/vitess@v0.9.0/go/sqltypes/result.go:170 +0x178
vitess.io/vitess/go/sqltypes.proto3ToRows(0xc0001be220, 0x4, 0x4, 0xc0001b4520, 0x1, 0x1, 0xc000cc7820, 0x415c78, 0x8)
```

**Comment by setassociative** [comment]
@indera-shsp yes! we've been running with that patch in prod since Feb 2019; it never made it upstream because, iirc, there was some concern that we were essentially trading one case of panic for another. A call out though is that this case is generally extremely transient -- if you're seeing it every time you try to connect to a vstream there is likely something else going on and it suggests that the connecting endpoint doesn't have the right schema.

My suggestion would be to drop into the Vitess slack and you can get some more targeted advice.
